### DISCUSSION TITLE
Generate json config file

### DISCUSSION THREAD

**Discussion by Elwell** [jvde-github/AIS-catcher#195] [Ideas]
Thanks for the addition of the json config, it does indeed make adding new feeder endpoints trivial!
However it took a bit of guessing and parsing the source to work out what all the possible options were nested under. Could you either include a full json config in the source (with all the possible options as default) or - and I suspect this is less likely to drift out of changes - have a way of generating a config file on the command line that we can alter.

Related to this, I've taken the [snippet of unit file](https://github.com/abcd567a/install-aiscatcher/blob/master/install-aiscatcher.sh#L77-L99) as a guide but changed 
`ExecStart=/usr/local/bin/AIS-Catcher -C /usr/local/etc/ais-catcher.json`

**Comment by jvde-github** [comment]
I have created an example config that should display most options that are available here: https://github.com/jvde-github/AIS-catcher#configuration-file

Can you elaborate a little bit what is missing and what you like to add?

Thanks, Jasper

**Comment by Elwell** [comment]
Thanks - I'd taken the "A fuller example config file looks as follows:" as a start with my rtl-sdr stick but couldn't work out what `"buffer_count":2` related to. I was hoping to see the metadata (again `"meta": "D"` took me some time to find as the correct syntax in the realtime stream on the web interface, but I'm not sure if this is only visible on the stdout stream (is that the one disabled by `"screen": 0` ?

Perhaps in the docs where you mention things like [Specifying Station Location](https://github.com/jvde-github/AIS-catcher#specifying-station-location) you could give the json alternative: _you need `"share_loc": true` under "server"_ for example. Similarly for things like enabling prometheus `"prome":true`, it's only that you mention it after the -N 8100 command line flag you realise it's part of the "server" section

Yours
Andrew

**Comment by jvde-github** [comment]
Thanks for sharing. I see where you are coming from now. There is indeed a grouping in settings as they are set on object level, e.g. a server, receiver, UDP stream, etc. The documentation is a bit silent on this as I thought it is only relevant to understand when running more complex configurations (like multiple receivers or webservers etc). I will see if I can bring this out more in the documentation.

meta is set for a receiver, so timestamps and signal strength are available in all output from this receiver (e.g. webserver, stdout, etc) as long as it is in the output format. If you have a second receiver (e.g. a second dongle) you need to set meta as well for that dongle.

Perhaps a diagram that shows how data flows though the program and how the settings will impact this will help as well.
